File: app/src/main/java/org/fattili/easypopup_demo/java/MainActivity.java
package org.fattili.easypopup_demo.java;

import android.os.Bundle;
import android.view.View;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.LifecycleOwner;

import org.fattili.easypopup.view.dialog.AlertDialogPop;
import org.fattili.easypopup_demo.R;

/**
 * 2021/3/29
 *
 * @author dengzhenli
 */
public class MainActivity extends AppCompatActivity implements LifecycleOwner {
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    private void topPop(View view) {
        new TopTestPop(this).register(this).show();
    }

    private void dialogPop(View view) {

        new AlertDialogPop.Builder(this)
                .setTitle("标题")
                .setMessage("是否启动自毁程序")
                .setMeasureButton(true, null, new View.OnClickListener() {
                    @Override
                    public void onClick(View v) {
                        finish();
                    }
                })
                .setCancelButton(true, null, null)
                .register(this)
                .show();
    }

    private void cardPop(View view) {
        new CardTestPop(this).register(this).show();
    }

    private void normalPop(View view) {
        new TestPop(this).register(this).show();
    }
}

File: app/src/main/java/org/fattili/easypopup_demo/kt/MainActivity.kt
package org.fattili.easypopup_demo.kt

import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.LifecycleOwner
import kotlinx.android.synthetic.main.pop_example.view.*
import org.fattili.easypopup.manager.EasyPopManager
import org.fattili.easypopup.view.EasyPop
import org.fattili.easypopup.view.dialog.AlertDialogPop
import org.fattili.easypopup_demo.R


class MainActivity : AppCompatActivity(), LifecycleOwner {

    var pop: EasyPop? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        EasyPopManager.register(this, this)
        pop = TestPop(this).show()
    }

    fun topPop(view: View) {
        TopTestPop(this).show()
    }

    fun dialogPop(view: View) {
        AlertDialogPop.Builder(this)
            .setTitle("标题")
            .setMessage("是否启动自毁程序")
            .setMeasureButton(true, null, View.OnClickListener { finish() })
            .setCancelButton(true, null, null)
            .show()
    }

    fun cardPop(view: View) {
        CardTestPop(this).show()
    }

    fun normalPop(view: View) {
        object : EasyPop(this@MainActivity) {
            override fun outClickable(): Boolean {
                return true
            }

            override fun initData() {}
            override fun initView(view: View?) {
                pop_example_text.text = "我是普通弹出窗"
            }

            override fun getLayoutId(): Int {
                return R.layout.pop_test
            }

        }.show()
    }

    override fun onWindowFocusChanged(hasFocus: Boolean) {
        super.onWindowFocusChanged(hasFocus)
        EasyPopManager.onWindowFocusChanged(this, hasFocus)
    }
}